projects/core/components/dialog/test/dialog.factory.spec.ts

```diff
@@ -0,0 +1,106 @@
+/* eslint-disable rxjs/no-ignored-observable */
+import type {tuiDialog, TuiDialogContext} from '@taiga-ui/core';
+import {EMPTY} from 'rxjs';
+
+const tuiDialogMock: typeof tuiDialog = jest.fn(() => jest.fn(() => EMPTY));
+
+{
+    // empty component
+    class TestComponent {}
+
+    const dialog = tuiDialogMock(TestComponent);
+
+    dialog().subscribe((_value: void) => {});
+    dialog(undefined).subscribe((_value: void) => {});
+    // @ts-expect-error TS2345: Argument of type `string` is not assignable to parameter of type `void`
+    dialog('test').subscribe((_value: void) => {});
+    // @ts-expect-error TS2769: Type `void` is not assignable to type `string`
+    dialog().subscribe((_value: string) => {});
+}
+
+{
+    // component with context
+    class TestComponent {
+        public readonly someContextProp!: TuiDialogContext<string, number>;
+    }
+
+    const dialog = tuiDialogMock(TestComponent);
+
+    dialog(123).subscribe((_value: string) => {});
+    // @ts-expect-error TS2554: Expected 1 arguments, but got 0
+    dialog();
```

Comment:
Probably, creating dialog without any `data` should be valid too 🤔 
Guys, what do you think about it ?

Reply:
You have a context with string data. So, you cannot create a dialog without passing a string to the data.

For dialogs that do not require input data, you must specify the data type as void or omit the entire context, as shown in the example above.